modules/cats-effect-io/src/main/scala/io/janstenpickle/trace4cats/cats/effect/DischargeSpan.scala

```diff
@@ -0,0 +1,34 @@
+package io.janstenpickle.trace4cats.cats.effect
+
+import cats.{Applicative, Monad}
+import cats.data.NonEmptyList
+import cats.effect.kernel.{Clock, Ref, Resource}
+import cats.syntax.all._
+import io.janstenpickle.trace4cats.kernel.{SpanCompleter, SpanSampler}
+import io.janstenpickle.trace4cats.model._
+import io.janstenpickle.trace4cats.{ErrorHandler, Span}
+
+case class DischargeSpan[F[_]: Monad: Clock: Ref.Make: TraceId.Gen: SpanId.Gen](
```

Comment:
@ybasket Why is `context` unused in the body?

Do you have a working example demonstrating `DischargeSpan`'s use-cases?

Reply:
@catostrophe `context` is part of `Span`'s signature, so we need one even if it's never used. And it may not be used in `child` because you would make everything a child of this one root, see also the sample branch below.

I created a sample application do demonstrate how the additions of this PR can be used in a dummy web service: https://github.com/ybasket/trace4cats-iolocal-demo. While the `main` branch demonstrates the usage of `DischargeSpan` and looks nice in the Jaeger UI, [a branch](https://github.com/ybasket/trace4cats-iolocal-demo/commit/ab220d4f9905d556f3e53b26afb013d7c801a713) demonstrates and explains the problems of using a classic root span to initialise the `IOLocal`. Some screenshots to illustrate that further:


<img width="1510" alt="Screenshot 2022-04-18 at 13 24 10" src="https://user-images.githubusercontent.com/2632023/163801530-6a404a85-60b3-4afe-87e2-9d00f09ebda8.png">
<img width="1124" alt="Screenshot 2022-04-18 at 13 24 18" src="https://user-images.githubusercontent.com/2632023/163801536-4732b5ac-ba1a-4fb3-a148-25a0fad3451f.png">

Ideally, some helpers would be added to make `DischargeSpan` & `EntryPoint` creation nicer (the snippet in `Main` of the demo service). But I think that belongs in a follow-up PR once we have the basic machinery merged.

Reply:
Ahh, sure, I missed that it's a part of the Span's signature. The demo looks great! It should be added to the examples when we finish.